numpy/typing/tests/data/reveal/shape.pyi

```diff
@@ -0,0 +1,20 @@
+from typing import Any, NamedTuple
+
+import numpy as np
+from typing_extensions import assert_type
+
+
+# Subtype of tuple[int, int]
+class XYGrid(NamedTuple):
+    x_axis: int
+    y_axis: int
+
+arr: np.ndarray[XYGrid, Any]
+
+# Test variance of _ShapeType_co
+def accepts_2d(a: np.ndarray[tuple[int, int], Any]) -> None: ...
+
+accepts_2d(arr)
```

Comment:
Since this doesn't "reveal" anything, it makes more sense to place this one in a `pass/shape.py`.

Alternatively, you could transform `accepts_2d` to e.g. return the shape of the input array, so that you could do `assert_type(accepts_2d(arr), XYGrid)`.